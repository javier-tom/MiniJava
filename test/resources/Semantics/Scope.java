class Main {
    public static void main(String[] args) {
        System.out.println((new Vars().run(10, 15)));
    }
}
class Vars {
    int a;
    int b;
    int c;
    int[] arr;
    boolean flag;

    public int run(int x, int y) {
        int[] arr3;
        z = 10;
        flag2 = false;
        a = arr[r];
        a = arr2[x];
        y = a + c;
        a = this.run2(a, b);
        arr3 = new int[2];
        arr[10000] = 777;
        c = arr.length;
        flag = arr3.length;
        return 0;
    }

    public int run2(int a, int b) {
        c = x + y;
        arr3[1] = 1;
        return 0;
    }
}