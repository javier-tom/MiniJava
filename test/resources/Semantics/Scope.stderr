Error on line 15: unknown variable z
Error on line 16: unknown variable flag2
Error on line 17: unknown variable r
Error on line 17: expected type int. Got type *error
Error on line 18: unknown variable arr2
Error on line 18: expected type int[]. Got type *error
Error on line 20: method whiles does not exist.
Error on line 24: cannot assign int to a boolean
Error on line 29: unknown variable x
Error on line 29: unknown variable y
Error on line 29: expected type int. Got type *error
Error on line 29: expected type int. Got type *error
Error on line 30: unknown variable arr3
Error on line 30: expected type int[]. Got type *error